tests/components/onvif/test_parsers.py

```diff
@@ -0,0 +1,271 @@
+    )
+
+
+async def test_tapo_vehicle(hass: HomeAssistant) -> None:
+    """Tests tns1:RuleEngine/TPSmartEventDetector/TPSmartEvent - vehicle."""
+    event = await get_event(
+        {
+            "Message": {
+                "_value_1": {
+                    "Data": {
+                        "ElementItem": [],
+                        "Extension": None,
+                        "SimpleItem": [{"Name": "IsVehicle", "Value": "true"}],
+                        "_attr_1": None,
+                    },
+                    "Extension": None,
+                    "Key": None,
+                    "PropertyOperation": "Changed",
+                    "Source": {
+                        "ElementItem": [],
+                        "Extension": None,
+                        "SimpleItem": [
+                            {
+                                "Name": "VideoSourceConfigurationToken",
+                                "Value": "vsconf",
+                            },
+                            {
+                                "Name": "VideoAnalyticsConfigurationToken",
+                                "Value": "VideoAnalyticsToken",
+                            },
+                            {"Name": "Rule", "Value": "MyTPSmartEventDetectorRule"},
+                        ],
+                        "_attr_1": None,
+                    },
+                    "UtcTime": datetime.datetime(
+                        2024, 11, 2, 0, 33, 11, tzinfo=datetime.UTC
+                    ),
+                    "_attr_1": {},
+                }
+            },
+            "ProducerReference": {
+                "Address": {
+                    "_attr_1": None,
+                    "_value_1": "http://192.168.56.127:5656/event",
+                },
+                "Metadata": None,
+                "ReferenceParameters": None,
+                "_attr_1": None,
+                "_value_1": None,
+            },
+            "SubscriptionReference": {
+                "Address": {
+                    "_attr_1": None,
+                    "_value_1": "http://192.168.56.127:2020/event-0_2020",
+                },
+                "Metadata": None,
+                "ReferenceParameters": None,
+                "_attr_1": None,
+                "_value_1": None,
+            },
+            "Topic": {
+                "Dialect": "http://www.onvif.org/ver10/tev/topicExpression/ConcreteSet",
+                "_attr_1": {},
+                "_value_1": "tns1:RuleEngine/TPSmartEventDetector/TPSmartEvent",
+            },
+        }
+    )
+
+    assert event is not None
+    assert event.name == "Vehicle Detection"
+    assert event.platform == "binary_sensor"
+    assert event.device_class == "motion"
+    assert event.value
+    assert (
+        event.uid
+        == f"{TEST_UID}_tns1:RuleEngine/TPSmartEventDetector/TPSmartEvent_VideoSourceToken_VideoAnalyticsToken_MyTPSmartEventDetectorRule"
+    )
```

Comment:
```suggestion
    assert event.uid == (
        f"{TEST_UID}_tns1:RuleEngine/LineDetector/"
        "Crossed_video_source_config1_analytics_video_source_MyLineDetectorRule"
    )
```
Please split long lines

Reply:
It looks like home assistant is using ruff, with the default `line-length` set to 88. I see `max-line-length-suggestions = 72` in `pyproject.toml`, but this doesn't seem to be used by ruff. It looks like that is only supported by pylint. The formatter doesn't touch the long strings it looks like, so I manually formatted the lines longer than 88.